chef/data_bags/crowbar/migrate/provisioner/024_add_autoyast_packages.rb

```diff
@@ -0,0 +1,11 @@
+def upgrade ta, td, a, d
+  a['suse'] ||= {}
+  a['suse']['autoyast'] ||= {}
+  a['suse']['autoyast']['packages'] ||= {}
+  return a, d
+end
+
+def downgrade ta, td, a, d
+  a['suse']['autoyast'].delete('packages')
```

Comment:
This could fail as `a['suse']` might not event exist. You could probably do something like:
`a.fetch('suse', {}).fetch('autoyast', {}).delete('packages')`